### PR TITLE
Vite migration

### LINKED ISSUE #76: Include social link to Monarch fediverse profile
Social links at the bottom left link to Medium, GH, and Twitter. Can these include Mastodon too?
https://genomic.social/@monarch_initiative

### LINKED ISSUE #79: Switch from Cypress to Playwright?
Playwright is a new and quickly-growing e2e testing library.

Cypress has been really good in my experience, but with some quirks, "obtuse-ness", and limitations. Playwright's tests look more like regular javascript and other (unit) tests, and do not require the odd chaining mechanism that Cypress does. Playwright can test more browsers, and more ruggedly hopefully due to the way it runs test with the new standard "Chrome Devtools Protocol" instead of Cypress's custom solution.

Deep comparison:
https://tweak-extension.com/blog/comparison-e2e-javascript-testing-frameworks

This would involve rewriting all the e2e tests, but I think it might be worth it in the long run. From everything I've read, I expect Playwright to get more and more popular, and Cypress less so. Also, there are tools that can help you automatically convert between the two:

https://twitter.com/housecor/status/1604970768659464193?s=20&t=8o68WWzQ_tDcFwkenJMg_A
https://demo.playwright.dev/cy2pw/

https://mtlynch.io/notes/cypress-vs-playwright/

### LINKED ISSUE #81: Switch from vue cli to vite?
https://cli.vuejs.org/

Vue CLI is now deprecated (sort of) 🥲 . For future maintainability, it might be best to switch the project over to Vite. Vite is now the recommended way to scaffold Vue projects.

With any luck, doing this transition would only take a day or so. Vite is much faster, cleaner, and easier to configure than Vue CLI. I've always had problems with Vue CLI anyway... verbose configuration, package compatibility issues, updating package issues, etc.

### PR DESCRIPTION
**Pull request by vincerubinetti**
roughly in order of commits and importance:

- migrate from vue-cli (deprecated) to vite (faster, more flexible, easier to manage, etc)
- switch from cypress to playwright (more familiar syntax, more features, faster, can run in parallel, can test more browsers, etc)
- switch from jest to vitest (a bit faster, otherwise the same, same syntax)
- move from jest-axe to playwright-axe (faster, more flexible, easier to run, etc)
- update and touchup tests during above
- move tests and fixtures up to top level with `/src` since they are shared across app, and makes tsconfig a little bit easier.

roughly in order of diff (excluding things above):

- update github actions workflows
- add vscode extension and settings recommendations
- rework fixtures index file. make easier matching with util func, make more robust fallback to make sure no real apis are ever hit.
- add "import-sorter" script and config to sort imports in all js/ts/vue files in consistent way
- update a few packages, e.g. font awesome and vue tippy
- fix bug: tab title doesn't update on initial page load
- typescript changes: interface to types for clarity, array and record to literals for brevity and familiarity, import types, remove separate component type files and merge into vue file
- export base urls for all api ts files
- add back in homepage demo mp4s and compress to reduce size
- fix tab component bugs: not redirecting on click, scroll resetting on click
- add mastodon link to footer and homepage
- remove export of sass vars to js
- rework unit test util funcs

Notes:
- locally on my m1, all the playwright tests (for 3 browsers) complete in < 20 seconds, but on github actions it takes ~2 minutes (for 1 browser). i dont think there's much we can do about this, as the free github actions machines simply aren't as powerful.

### CODE REVIEW

File: frontend/README.md
Comment (falquaddoomi) [inline frontend/README.md:1]: Seems like Vite might've nuked the original README; no worries if that's intentional, but I thought I'd mention it.

File: frontend/.vscode/settings.json
Comment (falquaddoomi) [inline frontend/.vscode/settings.json:1]: I usually gitignore IDE settings folders like `.vscode`, but perhaps you intended for everyone to have the same development setup. No worries if so, but I thought I'd bring it up.

**Review by falquaddoomi — COMMENTED** [review]
Looks good! Full disclosure, it seems there are a lot of import reorderings and other trivial typing changes that I skimmed over.

Regarding the tests: after cloning a fresh copy of the repo with your PR branch checked out, all the tests except `test:unit` run for me. This is the output I get from `yarn test:unit`; it's quite likely I'm doing something wrong since the GH action appears to run fine, but here it is anyway:

<details>
<summary>Log</summary>

```
% yarn test:unit
yarn run v1.22.19
warning ../../../../package.json: No license field
$ vitest run

 RUN  v0.29.8 /Users/faisala/Checkouts/for-reviews/monarch-app/frontend

stderr | unit/facets.test.ts > Query to params works
Couldn't map taxon label to id

stderr | unit/composables.test.ts > Statuses work
Error: dummy error
    at apiCall (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/unit/composables.test.ts:13:20)
    at query (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/util/composables.ts:57:22)

stderr | unit/uptime.test.ts > Help page renders uptimerobot statuses
ReferenceError: Request is not defined
    at Module.request (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/api/index.ts:56:19)
    at getUptimes (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/api/uptime.ts:36:20)
    at query (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/util/composables.ts:57:22)

stderr | unit/TheFeedbackForm.test.ts > Submits correctly when filled out
ReferenceError: Request is not defined
    at Module.request (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/api/index.ts:56:19)
    at Module.postFeedback (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/api/feedback.ts:28:16)
    at /Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/components/TheFeedbackForm.vue:176:16
    at query (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/util/composables.ts:57:22)
    at onSubmit (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/components/TheFeedbackForm.vue:138:5)

stderr | unit/composables.test.ts > Data and default value works
Error: dummy error
    at apiCall (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/unit/composables.test.ts:13:20)
    at query (/Users/faisala/Checkouts/for-reviews/monarch-app/frontend/src/util/composables.ts:57:22)

stderr | unit/composables.test.ts > Handles race conditions and calls success func
Stale query

 ❯ unit/facets.test.ts (2)
   ✓ Facets to filters works
   × Query to params works
 ✓ unit/AppSelectTags.test.ts (6) 1349ms
 ✓ unit/AppSelectMulti.test.ts (5) 605ms
 ✓ unit/AppTable.test.ts (6) 609ms
 ❯ unit/TheFeedbackForm.test.ts (1)
   × Submits correctly when filled out
 ❯ unit/uptime.test.ts (1)
   × Help page renders uptimerobot statuses
 ✓ unit/AppSelectAutocomplete.test.ts (3)
 ✓ unit/composables.test.ts (4)
 ❯ unit/api.test.ts (12)
   × Post feedback requests correctly
   × Node search returns correctly
   × Get node associations requests correctly
   × Get gene requests correctly
   × Get hierarchy requests correctly
   × Get publication requests correctly
   × Phenotype explorer requests correctly
   × Get entity requests correctly
   × Get entity returns correct fixture data
   × Get HistoPheno requests correctly
   × Get autocomplete request correctly
   × Get search results request correctly
 ✓ unit/AppTabs.test.ts (3)
 ✓ unit/AppSelectSingle.test.ts (3)
 ✓ unit/dom.test.ts (1)
 ✓ unit/AppHeading.test.ts (2)
 ✓ unit/AppTextbox.test.ts (1)
 ✓ unit/AppAccordion.test.ts (1)
 ✓ unit/AppMarkdown.test.ts (1)
 ✓ unit/AppLink.test.ts (2)
 ✓ unit/AppButton.test.ts (2)
 ✓ unit/object.test.ts (1)
 ✓ unit/url.test.ts (1)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 15 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  unit/TheFeedbackForm.test.ts > Submits correctly when filled out
AssertionError: expected false to be truthy
 ❯ unit/TheFeedbackForm.test.ts:26:25
     24|   /** test status message and expect to be success */
     25|   const link = wrapper.find(".status a");
     26|   expect(link.exists()).toBeTruthy();
       |                         ^
     27| });
     28| 

Could not display diff. It's possible objects are too large to compare.
Try increasing --outputDiffMaxSize option.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/15]⎯

 FAIL  unit/api.test.ts > Post feedback requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.postFeedback src/api/feedback.ts:28:16
 ❯ unit/api.test.ts:25:3

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/15]⎯

 FAIL  unit/api.test.ts > Node search returns correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getSearchResults src/api/node-search.ts:60:26
 ❯ unit/api.test.ts:70:38

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/15]⎯

 FAIL  unit/api.test.ts > Get node associations requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getTabulatedAssociations src/api/node-associations.ts:97:26
 ❯ unit/api.test.ts:78:3

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/15]⎯

 FAIL  unit/api.test.ts > Get gene requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getGene src/api/genes.ts:51:26
 ❯ unit/api.test.ts:98:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/15]⎯

 FAIL  unit/api.test.ts > Get hierarchy requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getHierarchy src/api/node-hierarchy.ts:74:26
 ❯ unit/api.test.ts:104:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/15]⎯

 FAIL  unit/api.test.ts > Get publication requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getSummaries src/api/publications.ts:57:28
 ❯ unit/api.test.ts:115:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/15]⎯

 FAIL  unit/api.test.ts > Phenotype explorer requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.compareSetToSet src/api/phenotype-explorer.ts:124:26
 ❯ unit/api.test.ts:120:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[8/15]⎯

 FAIL  unit/api.test.ts > Get entity requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getEntity src/api/entity.ts:7:32
 ❯ unit/api.test.ts:142:9

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[9/15]⎯

 FAIL  unit/api.test.ts > Get entity returns correct fixture data
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getEntity src/api/entity.ts:7:32
 ❯ unit/api.test.ts:149:24

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[10/15]⎯

 FAIL  unit/api.test.ts > Get HistoPheno requests correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getHistoPheno src/api/histopheno.ts:6:26
 ❯ unit/api.test.ts:157:28

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[11/15]⎯

 FAIL  unit/api.test.ts > Get autocomplete request correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getAutocomplete src/api/search.ts:16:41
 ❯ unit/api.test.ts:163:30

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[12/15]⎯

 FAIL  unit/api.test.ts > Get search results request correctly
ReferenceError: Request is not defined
 ❯ Module.request src/api/index.ts:56:19
     54| 
     55|   /** make request object */
     56|   const request = new Request(url, options);
       |                   ^
     57| 
     58|   /** first check if request is cached */
 ❯ Module.getSearchResults src/api/node-search.ts:60:26
 ❯ unit/api.test.ts:168:25

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[13/15]⎯

 FAIL  unit/facets.test.ts > Query to params works
AssertionError: expected { taxon: [ 'mus musculus' ] } to strictly equal { taxon: [ 'NCBITaxon:10090' ] }
 ❯ unit/facets.test.ts:68:5
     66|       }
     67|     )
     68|   ).toStrictEqual({ taxon: ["NCBITaxon:10090"] });
       |     ^
     69| });
     70| 

  - Expected  - 1
  + Received  + 1

    Object {
      "taxon": Array [
  -     "NCBITaxon:10090",
  +     "mus musculus",
      ],
    }

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[14/15]⎯

 FAIL  unit/uptime.test.ts > Help page renders uptimerobot statuses
AssertionError: expected false to be truthy
 ❯ unit/uptime.test.ts:17:58
     15|    * displayed as appropriate status components.
     16|    */
     17|   expect(wrapper.find("[data-code='success']").exists()).toBeTruthy();
       |                                                          ^
     18|   expect(wrapper.find("[data-code='error']").exists()).toBeTruthy();
     19|   expect(wrapper.find("[data-code='paused']").exists()).toBeTruthy();

Could not display diff. It's possible objects are too large to compare.
Try increasing --outputDiffMaxSize option.

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[15/15]⎯

 Test Files  4 failed | 16 passed (20)
      Tests  15 failed | 43 passed (58)
   Start at  11:54:38
   Duration  5.73s (transform 233ms, setup 28.71s, collect 222ms, tests 4.05s)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>

**Comment by vincerubinetti** [comment]
Not sure and can't check because at lunch atm, but you might need to upgrade your version of node. Which version are you using? 

I believe one of the recent versions added native support for things like fetch and request, without needing a shim, and i'm relying on that. That's why i removed the node-fetch dev dependency for example.

**Comment by falquaddoomi** [comment]
> Not sure and can't check because at lunch atm, but you might need to upgrade your version of node. Which version are you using?

I'm currently using Node v16.14.0, but I can switch to whichever version you'd suggest.

**Comment by vincerubinetti** [comment]
Could you try 18, i think that may have been the version that introduced it and i'd like to make sure that this is indeed the problem. If its too much trouble, when i get back i can use nvm to easily revert to 16 and try it out.

**Comment by falquaddoomi** [comment]
> Could you try 18, i think that may have been the version that introduced it and i'd like to make sure that this is indeed the problem. If its too much trouble, when i get back i can use nvm to easily revert to 16 and try it out.

Looks like that fixed it! I used nvm to switch to v18.16.0 and `yarn test:unit` runs all the tests successfully.

(Also, I see now that the node version is defined in the GH test-frontend action; I'll just check there next time and not bother you about it.)

**Comment by vincerubinetti** [comment]
@kevinschaper @putmantime @glass-ships You may want to look at the bullet points for this PR just to maintain familiarity. On the whole, it's mostly configuration changes and simple find/replace type changes. Some very minor bug fixes and additions. Maybe one thing to note is the Axe tests moving from jest to playwright.

**Comment by kevinschaper** [comment]
Probably we need to update the Makefile commands for the frontend, it's failing, but since it's not updated, I bet it's not the right set of commands. Is just removing --detectOpenHandles enough?

This is what's in there right now for testing the front-end code
```
.PHONY: test-frontend
test-frontend: install-frontend model
        cd frontend && \
                npx update-browserslist-db@latest && \
                yarn build && \
                yarn test:unit --detectOpenHandles  # && \
                # yarn test:lint && \
                # yarn test:axe && \
                # yarn test:e2e && \
```

and it fails with: 

```
dist/assets/shahim-essaid-e7142e7c.js                               4.57 kB │ gzip:   3.40 kB
dist/assets/bgee-03652309.js                                        5.17 kB │ gzip:   3.89 kB
dist/assets/reactome-e6490330.js                                    5.51 kB │ gzip:   4.21 kB
dist/assets/index-4aa439ba.js                                   1,907.02 kB │ gzip: 591.48 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 6.18s
✨  Done in 6.44s.
yarn run v1.22.19
$ vitest run --detectOpenHandles
file:///Users/kschaper/Documents/Monarch/monarch-app/frontend/node_modules/cac/dist/index.mjs:400
          throw new CACError(`Unknown option \`${name.length > 1 ? `--${name}` : `-${name}`}\``);
                ^

CACError: Unknown option `--detectOpenHandles`
    at Command.checkUnknownOptions (file:///Users/kschaper/Documents/Monarch/monarch-app/frontend/node_modules/cac/dist/index.mjs:400:17)
    at CAC.runMatchedCommand (file:///Users/kschaper/Documents/Monarch/monarch-app/frontend/node_modules/cac/dist/index.mjs:598:13)
    at CAC.parse (file:///Users/kschaper/Documents/Monarch/monarch-app/frontend/node_modules/cac/dist/index.mjs:537:12)
    at file:///Users/kschaper/Documents/Monarch/monarch-app/frontend/node_modules/vitest/dist/cli.js:67:5
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

```

**Comment by vincerubinetti** [comment]
I was going to ask about those. I don't understand why they are there. Any time a PR is opened, the frontend will be tested and built and deployed with the workflows that I've written. None of the frontend related commands in that makefile should be necessary, except maybe the `dev-frontend` one which should be changed to `yarn dev`.

**Comment by vincerubinetti** [comment]
Well, it's not the typical way a frontend dev would interact with and run things, but I suppose some other people on the team might use the makefile instead of directly calling the package.json scripts with yarn. So I've updated the makefile.

I'm also going to take a stab at rewriting the root readme to reflect the fact that this is now a monorepo.

**Comment by kevinschaper** [comment]
Thanks!  yeah, we've tried to make it a standard across repos as a kind of executable documentation, but it's also nice to be able to run `make test` from the top and have both sets of tests run - especially after a model change which touches the generated code on both sides.

**Comment by vincerubinetti** [comment]
Do you want to try to figure out how to move the docs to `/docs` in this PR? I tried looking into this command `mkdocs gh-deploy`, but couldn't immediately see any options to deploy it to a subfolder.

File: frontend/.vscode/settings.json
Comment (vincerubinetti) [reply frontend/.vscode/settings.json:1]: Yeah in this case, a few people have asked about having the right setup. Also there can be some hard to resolve typescript/eslint/etc warnings if you don't have the right extensions. I also hand wrote these recommendation extensions/settings files, so there's only what's necessary in there.

File: frontend/README.md
Comment (vincerubinetti) [reply frontend/README.md:1]: I've now copied in the useful parts from the old readme and updated it.

**Review by falquaddoomi — APPROVED** [review]

**Comment by vincerubinetti** [comment]
@kevinschaper See the latest commit. I believe this will make your docs deploy to a `/docs` folder in `gh-pages`. And then, my deploy process for the frontend basically also deploys to `gh-pages` _around_ that, i.e. replaces all the contents on each build, except for `/docs` which it leaves alone.

This would also make your docs pages be at `monarch-app.monarchinitiative.org/docs` (we're setting up the `monarch-app` subdomain right now to match the repo name).

**Comment by vincerubinetti** [comment]
Not sure what's going on with the Netlify PR preview detecting no changes and cancelling the build, but we'll figure that out next PR.

**Pull request merged**